frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import Draggable, { DraggableData, DraggableBounds } from 'react-draggable';
+import { useBounds, getMastheadHeight, dragHandle } from './drawer-utils';
+import './Drawer.scss';
+
+type DrawerProps = {
+  height?: number;
+  toggle: boolean;
+  maxHeight?: number;
+  minHeight?: number;
+  initialHeight?: number;
+  isResizable?: boolean;
+  bounds?: DraggableBounds | string;
+  onResize?: (e: MouseEvent, data: DraggableData) => void;
+  onResizeStart?: (e: MouseEvent, data: DraggableData) => void;
+  onResizeStop?: (e: MouseEvent, data: DraggableData) => void;
+};
+
+const DEFAULT_HEIGHT = 500;
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  initialHeight = DEFAULT_HEIGHT,
+  height = initialHeight,
+  maxHeight,
+  bounds,
+  toggle,
+  minHeight,
+  isResizable = false,
+  onResize,
+  onResizeStart,
+  onResizeStop,
+}) => {
+  const [resizeHeight, setResizeHeight] = React.useState(!isResizable ? initialHeight : 0);
+  const defaultBound = useBounds(height, minHeight);
+  const mastHeadHeight = getMastheadHeight();
+  const handleDrag = (e: MouseEvent, data: DraggableData) => {
+    onResize ? onResize(e, data) : setResizeHeight(resizeHeight - data.y);
+  };
+  const draggable = (
+    <Draggable
+      axis="y"
+      handle="#header-drag"
+      bounds={bounds || defaultBound}
+      onDrag={handleDrag}
+      onStart={onResizeStart}
+      onStop={onResizeStop}
```

Comment:
This component should be taking care of all the quirks around dragging and computing open and closed state along with notifying height changes.